src/Microsoft.OData.Core/Json/JsonValueUtilsAsync.cs

```diff
@@ -122,27 +121,26 @@ internal static Task WriteValueAsync(this TextWriter writer, long value)
         /// </summary>
         /// <param name="writer">The text writer to write the output to.</param>
         /// <param name="value">Double value to be written.</param>
-        internal static async Task WriteValueAsync(this TextWriter writer, double value)
+        internal static Task WriteValueAsync(this TextWriter writer, double value)
         {
             Debug.Assert(writer != null, "writer != null");
 
             if (JsonSharedUtils.IsDoubleValueSerializedAsString(value))
             {
-                await writer.WriteQuotedAsync(value.ToString(ODataNumberFormatInfo)).ConfigureAwait(false);
+                return writer.WriteQuotedAsync(value.ToString(ODataNumberFormatInfo));
             }
-            else
-            {
-                // double.ToString() supports a max scale of 14,
-                // whereas double.MinValue and double.MaxValue have 16 digits scale. Hence we need
-                // to use XmlConvert in all other cases, except infinity
-                string valueToWrite = XmlConvert.ToString(value);
 
-                await writer.WriteAsync(valueToWrite).ConfigureAwait(false);
-                if (valueToWrite.IndexOfAny(DoubleIndicatingCharacters) < 0)
-                {
-                    await writer.WriteAsync(".0").ConfigureAwait(false);
-                }
+            // double.ToString() supports a max scale of 14,
+            // whereas double.MinValue and double.MaxValue have 16 digits scale. Hence we need
+            // to use XmlConvert in all other cases, except infinity
+            var valueToWrite = XmlConvert.ToString(value);

```

Comment:
>var [](http://example.com/codeflow?start=12&length=3)

is there a reason you use var rather than string? convention in the repo is to avoid var (except in some tests).

Reply:
@mikepizzo I've changed the declaration